Kitodo/src/main/java/de/sub/goobi/helper/ProjectHelper.java

```diff
@@ -39,157 +34,92 @@
+        // processes according to its title
+        // the position will be calculated by the average of 'ordering' of
         // steps
 
-        list = critSteps.list();
-
         String title;
-        Double averageStepOrder;
-        Long numberOfSteps;
+        Double averageTaskOrder;
+        Long numberOfTasks;
         Long numberOfImages;
 
         List<StepInformation> workFlow = new ArrayList<>();
 
-        for (Object obj : list) {
-            Object[] row = (Object[]) obj;
-
-            title = (String) (row[FieldList.stepName.fieldLocation]);
-            numberOfSteps = (Long) (row[FieldList.stepCount.fieldLocation]);
-            averageStepOrder = (Double) (row[FieldList.stepOrder.fieldLocation]);
+        for (int i = 0; i < tasks.size(); i++) {
+            title = tasks.get(i).getTitle();
+            numberOfTasks = tasksSize.get(i);
+            averageTaskOrder = averageTaskOrders.get(i);
 
             // in this step we only take the steps which are present in each of
             // the workflows
-            if (numberOfSteps.equals(totalNumberOfProc)) {
-                StepInformation newStep = new StepInformation(title, averageStepOrder);
+            if (numberOfTasks.equals(totalNumberOfProcesses)) {
+                StepInformation newStep = new StepInformation(title, averageTaskOrder);
                 newStep.setNumberOfTotalImages(totalNumberOfImages.intValue());
-                newStep.setNumberOfTotalSteps(totalNumberOfProc.intValue());
+                newStep.setNumberOfTotalSteps(totalNumberOfProcesses.intValue());
                 workFlow.add(newStep);
             }
         }
 
-        Criteria critStepDone = session.createCriteria(Task.class, "step");
-
-        critStepDone.createCriteria("process", "proc");
-
-        critStepDone.add(Restrictions.eq("step.processingStatus", TaskStatus.DONE.getValue()));
-        critStepDone.add(Restrictions.eq("proc.template", Boolean.FALSE));
-        critStepDone.add(Restrictions.eq("proc.project", project));
-
-        ProjectionList proCount = Projections.projectionList();
-
-        proCount.add(Projections.groupProperty(("step.title")));
-        proCount.add(Projections.count("proc.id"));
-        proCount.add(Projections.sum("proc.sortHelperImages"));
-
-        critStepDone.setProjection(proCount);
-
-        list = critStepDone.list();
-
-        for (Object obj : list) {
-
-            Object[] row = (Object[]) obj;
-
-            title = (String) (row[FieldList.stepName.fieldLocation]);
-            numberOfSteps = (Long) (row[FieldList.stepCount.fieldLocation]);
-            numberOfImages = (Long) (row[FieldList.imageCount.fieldLocation]);
+        List<Task> doneTasks = serviceManager.getTaskService()
+                .getTasksWithProcessingStatusForProjectHelper(
+                        TaskStatus.DONE.getValue(), project.getId());
+        List<Long> doneTasksSize = serviceManager.getTaskService()
```

Comment:
The amount of `doneTasks` and the number in `doneTaskSize` is only equal if there no data inserted between both calls. Value of `doneTaskSize` should be gathered from  `doneTasks.size()` call instead of quering the database again.